src/components/merchandise/MerchandisePaymentButton.tsx

```diff
@@ -122,7 +131,14 @@ const GeneralMerchandisePaymentBlock: React.VFC<{
           colorScheme="primary"
           isFullWidth
           isDisabled={merchandise.isLimited && (quantity === 0 || quantity > remainQuantity)}
-          onClick={() => quantity && handleClick().then(() => history.push('/cart'))}
+          onClick={() => {
+            const resource = resourceCollection.filter(notEmpty)[0]
```

Comment:
why not use [`find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) here ?

Reply:
using `filter` will run into ts bug, noUncheckedIndexedAccess.
so `find` is better.